go/src/koding/klient/app/defertime.go

```diff
@@ -0,0 +1,82 @@
+package app
+
+import "time"
+
+// DeferTime allows to invoke registered function after provided duration. It
+// can cancel deferred function call even if internal timer was already started.
+type DeferTime struct {
```

Comment:
`time.AfterFunc` starts the timer immediately after called, the returned timer will still need logic for its cancellation and re-initialization.